src/analyz/to_json.rs

```diff
@@ -482,16 +501,18 @@ impl ToJson<'_> for BinOp {
 impl ToJson<'_> for CastKind {
     fn to_json(&self, _: &mut MirState) -> serde_json::Value {
         match self {
-            CastKind::PointerExposeAddress => json!({ "kind": "PointerExposeAddress" }),
-            CastKind::PointerFromExposedAddress => json!({ "kind": "PointerFromExposedAddress" }),
-            CastKind::DynStar => json!({ "kind": "DynStar" }),
+            CastKind::PointerExposeProvenance => json!({ "kind": "PointerExposeProvenance" }),
+            CastKind::PointerWithExposedProvenance => json!({ "kind": "PointerWithExposedProvenance" }),
             CastKind::IntToInt => json!({ "kind": "IntToInt" }),
             CastKind::FloatToInt => json!({ "kind": "FloatToInt" }),
             CastKind::FloatToFloat => json!({ "kind": "FloatToFloat" }),
             CastKind::IntToFloat => json!({ "kind": "IntToFloat" }),
             CastKind::PtrToPtr => json!({ "kind": "PtrToPtr" }),
             CastKind::FnPtrToPtr => json!({ "kind": "FnPtrToPtr" }),
-            CastKind::Pointer(_) => json!({ "kind": format!("{:?}", self) }),
+            CastKind::Transmute => json!({ "kind": "Transmute" }),
+
+            // Data-carrying variants — use Debug formatting
+            CastKind::PointerCoercion { .. } => json!({ "kind": format!("{:?}", self) }),
```

Comment:
Upon further thought, I'm not quite sure that this is right. `PointerCoercion` has fields of type `PointerCoercion` and `CoercionSource`, which we likely want to serialize to JSON rather than relying on their `Debug` output.